.github/workflows/amplify_api.yaml

```diff
@@ -0,0 +1,26 @@
+name: amplify_api
+on:
+  push:
+    branches:
+      - main
+      - stable
+      - next
+  pull_request:
+    paths:
+      - 'packages/api/amplify_api/**/*.dart'
```

Comment:
question: Is the reason for having the category flutter vm tests all be their own workflow that is seems difficult to enforce this pr/pattern thing any other way? Absent that, seems like a matrix would work nicely but not sure how you would do this via matrix. Seems to make sense, just trying to make sure I understand why.

Reply:
Not sure I understand the question. Each package gets its own workflow. Which workflows run depends on the structure of the package. The `flutter_vm` workflow is chosen for any package which depends on Flutter.

Reply:
My bad for confusing phrasing (I also saw this bit/left this comment before I saw the autogeneration code). My point was that the similar workflow logic is running/repeated across these categories. You could run them via a matrix to save repitition like the integration tests but I'm not sure how you would stop one category from running on pushes to certain files with that pattern, so I thought that was why you did it this way. Knowing they are autogenerated would also be another reason.

Reply:
Yeah, since paths affect the workflow as a whole and there isn't a way to control this at any lower level from what I've seen. Since it's autogenerated, there's no real maintenance risk to 1 workflow vs. 50. It's also why all the workflow logic is centralized in `flutter_vm`, so all these files do is control the `paths` to run on, essentially.

We also have more information about the packages in `aft` than we do once the workflow is running which is another reason we generate them.